src/v/cluster/client_quota_backend.cc

```diff
@@ -0,0 +1,43 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "cluster/client_quota_backend.h"
+
+#include "cluster/client_quota_store.h"
+#include "cluster/commands.h"
+#include "cluster/controller_snapshot.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/util/variant_utils.hh>
+
+namespace cluster::client_quota {
+
+ss::future<std::error_code> backend::apply_update(model::record_batch batch) {
+    auto cmd = co_await cluster::deserialize(std::move(batch), commands);
+    co_await _quotas.invoke_on_all([&cmd](store& quota_store) {
```

Comment:
question: maybe a dumb one, but is there a race condition here?  is it possible for multiple shards to call `apply_update` at the same time causing the `invoke_on_all` to not apply the command correctly?

Reply:
This is a great question. The reason why there isn't a race condition here is that there is only a single `backend` instance and a single `controller_stm` and updates are applied sequentially.
https://github.com/redpanda-data/redpanda/blob/f0e5e83e2485c8448811c19db9c6aa9a193fc336/src/v/cluster/controller.cc#L273

I am not sure why we construct them as `ss::sharded<>` services, but we only create a single instead of all the backends and the controller stm.